rust/src/tests/errors/enotdir.rs

```diff
@@ -0,0 +1,36 @@
+/// Asserts that it returns ENOTDIR if a component of the path prefix is not a directory.
+macro_rules! assert_enotdir_comp {
```

Comment:
The name of this macro suggests that it expands to a statement, something similar to `assert!`.  But it actually expands to an item.  I suggest renaming it to something like `enotdir_test_case!`.